File: common/src/main/java/com/pranav/common/Indexer.java
package com.pranav.common;

import com.pranav.common.util.FileUtil;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.io.IOException;

public class Indexer {

    private JSONObject json;

    private String filePath;

    public Indexer(String fileName) throws JSONException {
        filePath = FileUtil.getCacheDir() + fileName + ".json";
        load();
    }

    public void load() throws JSONException {
        var indexFile = new File(filePath);
        try {
            if (!indexFile.exists()) {
                FileUtil.writeFile(filePath, "{}");
            }
            var index = FileUtil.readFile(indexFile);
            json = new JSONObject(index);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public Indexer put(String key, String value) throws JSONException {
        json.put(key, value);
        return this;
    }

    public Indexer put(String key, long value) throws JSONException {
        json.put(key, value);
        return this;
    }

    public boolean notHas(String key) throws JSONException {
        return !json.has(key);
    }

    public String getString(String key) throws JSONException {
        return json.getString(key);
    }

    public long getLong(String key) {
        try {
            return json.getLong(key);
        } catch (JSONException e) {
            return 0;
        }
    }

    public String asString() throws JSONException {
        return json.toString(4);
    }

    public void flush() {
        try {
            FileUtil.writeFile(filePath, asString());
        } catch (Exception e) {
        }
    }
}

File: common/src/main/java/com/pranav/common/Indexer.kt
package com.pranav.common

import com.pranav.common.util.FileUtil

import org.json.JSONException
import org.json.JSONObject

import java.io.File
import java.io.IOException

class Indexer {

    private lateinit var json: JSONObject

    private var filePath: String

    @Throws(JSONException::class)
    constructor(projectName: String, projectCacheDirPath: String) {
        filePath = projectCacheDirPath + projectName + ".json"
        load()
    }

    @Throws(JSONException::class)
    fun load()  {
        val indexFile = File(filePath)
        try {
            if (!indexFile.exists()) {
                FileUtil.writeFile(filePath, "{}")
            }
            val index = FileUtil.readFile(indexFile)
            json = JSONObject(index)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    @Throws(JSONException::class)
    fun put(key: String, value: String) : Indexer {
        json.put(key, value)
        return this
    }

    @Throws(JSONException::class)
    fun put(key: String, value: Long) : Indexer {
        json.put(key, value)
        return this
    }

    @Throws(JSONException::class)
    fun notHas(key: String) : Boolean {
        return !json.has(key)
    }

    @Throws(JSONException::class)
    fun getString(key: String) : String {
        return json.getString(key)
    }

    fun getLong(key: String) : Long {
        try {
            return json.getLong(key)
        } catch (e: JSONException) {
            return 0
        }
    }

    @Throws(JSONException::class)
    fun asString() : String {
        return json.toString(4)
    }

    fun flush() {
        try {
            FileUtil.writeFile(filePath, asString())
        } catch (e: Exception) {
        }
    }
}
